src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends T> type);
```

Comment:
Do not use raw types. The return type of `getAll` should be parameterized to match the type parameter of the `MachineService`. Consider changing the return type to `List<T>`.